### ISSUE TITLE
[android] Java.Net.SocketException and Javax.Net.Ssl.SSLException are thrown in Debug not in Release

### ISSUE THREAD

**Issue by gsgou** [tranb3r/securehttpclient#15]
I see in code that you correctly catch Java exceptions and re-throw them as HttpRequestExceptions.
Strangely when i am in Debug mode only i catch no HttpRequestException instead the Java ones as
unhandled exceptions.

Its an edge-case but maybe if you have time you can have a look at it.
Glad to help when a time-window opens.

```
var innerHandler = new SecureHttpClientHandler(null);
client = new HttpClient(innerHandler);
```
at Java.Interop.JniEnvironment+InstanceMethods.CallObjectMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x0006e] in <1959115d56f8444789986cf39185638c>:0 
  at Android.Runtime.JNIEnv.CallObjectMethod (System.IntPtr jobject, System.IntPtr jmethod, Android.Runtime.JValue* parms) [0x00000] in /Users/builder/azdo/_work/2/s/xamarin-android/src/Mono.Android/Android.Runtime/JNIEnv.g.cs:102 
  at Square.OkHttp3.IInterceptorChainInvoker.Proceed (Square.OkHttp3.Request request) [0x0005a] in <0eabf675945747edb17222b3720748da>:0 
  at SecureHttpClient.DecompressInterceptor.Intercept (Square.OkHttp3.IInterceptorChain chain) [0x00007] in <ab31e2c702f84ed6a7b15b1e759c9350>:0 
  at Square.OkHttp3.IInterceptorInvoker.n_Intercept_Lokhttp3_Interceptor_Chain_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_chain) [0x0000f] in <0eabf675945747edb17222b3720748da>:0 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.115(intptr,intptr,intptr)
  --- End of managed Javax.Net.Ssl.SSLException stack trace ---
javax.net.ssl.SSLException: Write error: ssl=0x6ecedf6d08: I/O error during system call, Software caused connection abort
	at com.android.org.conscrypt.NativeCrypto.SSL_write(Native Method)
	at com.android.org.conscrypt.NativeSsl.write(NativeSsl.java:426)
	at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLOutputStream.write(ConscryptFileDescriptorSocket.java:626)
	at okio.OutputStreamSink.write(JvmOkio.kt:56)
	at okio.AsyncTimeout$sink$1.write(AsyncTimeout.kt:99)
	at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.kt:255)
	at okio.RealBufferedSink.write(RealBufferedSink.kt:146)
	at okhttp3.internal.http1.Http1ExchangeCodec$KnownLengthSink.write(Http1ExchangeCodec.kt:271)
	at okio.ForwardingSink.write(ForwardingSink.kt:29)
	at okhttp3.internal.connection.Exchange$RequestBodySink.write(Exchange.kt:218)
	at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.kt:255)
	at okio.RealBufferedSink.write(RealBufferedSink.kt:185)
	at okhttp3.RequestBody$Companion$toRequestBody$2.writeTo(RequestBody.kt:152)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:59)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at crc647d92411772b9fe65.DecompressInterceptor.n_intercept(Native Method)
	at crc647d92411772b9fe65.DecompressInterceptor.intercept(DecompressInterceptor.java:30)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

**Comment by tranb3r** [comment]
Yes, I'm aware of it. I think it's related to the way xamarin handles exceptions and pass then between java and .net, which is different between debug and release. But so far I haven't found a proper explanation and a way to fix it.

However, I was assuming this issue is not occurring when using the SecureHttpClient nuget, right ? (since it's built in release mode).
Could you explain what is your scenario ?
- are you debugging an application that uses the SecureHttpClient nuget ?
- or are you debugging the SecureHttpCient code ?

**Comment by gsgou** [comment]
i am deubbing an application that uses the SecureHttpClient nuget.

**Comment by gsgou** [comment]
I will try it and report to you. Also i noticed in AppCenter logs that i do not have a single HttpRequestException.
Since you catch some Java exceptions and rethrow them as HttpRequestExceptions i should at least have one
instead of Java ones. This is not the case.

**Comment by tranb3r** [comment]
Fixed in 1.18.4

**Comment by tranb3r** [comment]
> Could you add a catch also for Java.Net.ConnectException when there is no internet connection

@gsgou Could you please open a new issue ? I'm not sure I understand your request, so it's better to separate discussions.
